spring-cloud-stream-integration-tests/src/test/java/org/springframework/cloud/stream/config/LegacyContentTypeTests.java

```diff
@@ -54,16 +55,15 @@ public void testOriginalContentTypeIsRetrievedForLegacyContentHeaderType() throw
 		MessageHandler messageHandler = new MessageHandler() {
 			@Override
 			public void handleMessage(Message<?> message) throws MessagingException {
-				assertThat(message.getPayload()).isInstanceOf(String.class);
-				assertThat(message.getPayload()).isEqualTo("{\"message\":\"Hi\"}");
+				assertThat(message.getPayload()).isInstanceOf(byte[].class);
+				assertThat(new String(((byte[])message.getPayload()), StandardCharsets.UTF_8)).isEqualTo("{\"message\":\"Hi\"}");
```

Comment:
Hmmm - I really don't think we need all this charset noise in these tests.

Reply:
in my "previous life in other projects", I've encountered occasional test failures on Windows machines whenever charset was omitted. So this may be a bit of a paranoia on my end